config/routes.rb

```diff
@@ -75,6 +75,10 @@
     end
   end
 
+  get '/unsubscribe/:id/:barcode' => 'unsubscribe#show',  as: 'unsubscribe'
+
+  delete 'unsubscribe/:id/:barcode' => 'unsubscribe#delete', as: 'unsubscribe_delete'
```

Comment:
I would put these as a post 'unsubscribe' as a registration collection member. The route then becomes /event/event_id/registration/unsubscribe with a barcode as post param.